__tests__/app.test.js

```diff
@@ -184,16 +184,91 @@ describe('/api/articles/:article_id/comments' , () => {
+            .then(({body: {comments}}) => {
+                expect(comments).toEqual([])
+            })
+        })
+        test('GET: 200 - returns an empty array when the article exists but has no comments' , () => {
+            return request(app)
+            .get('/api/articles/7/comments')
+            .expect(200)
+            .then(({body: {comments}}) => {
+                expect(comments).toEqual([])
+            })
+        })
+    })
+    describe('POST' , () => {
+        test('POST: 201 - returns the comment object the has just been posted' , () => {
+            const postInfo = {username: 'butter_bridge' , body: 'Me too!'}
+            return request(app)
+            .post('/api/articles/1/comments')
+            .send(postInfo)
+            .expect(201)
+            .then(({body: {comment}}) => {
+                expect(comment).toEqual({
+                    comment_id: 19,
+                    votes: 0,
+                    created_at: expect.any(String),
+                    author: 'butter_bridge',
+                    body: 'Me too!',
+                    article_id: 1
+                })
+            })
+        })
+        test('POST: 400 - Bad request returns an error when the username is not valid i.e not in the users table' , () => {
+            const postInfo = {username: 'invalidUsername468' , body: 'hi'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(postInfo)
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe('400 - Bad Request')
+            })
+        })
+        test('POST: 400 - Bad request returns an error when a user tries to comment on an article that does not exist' , () => {
+            const postInfo = {username: 'butter_bridge' , body: 'omg'}
+            return request(app)
+            .post('/api/articles/725/comments')
+            .send(postInfo)
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe('400 - Bad Request')
+            })
+        })
+        test('POST: 400 - Bad request returns an error when the request body does not have the correct format' , () => {
+            const postInfo = {wrongKey: 2}
+            return request(app)
+            .post('/api/articles/3/comments')
+            .send(postInfo)
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe('400 - Bad Request Incorrect Format')
+            })
+        })
+        test('POST: 400 - Bad request returns an error when a user tries to comment on an article id that is not a number' , () => {
+            const postInfo = {username: 'butter_bridge' , body: 'omg'}
+            return request(app)
+            .post('/api/articles/hello/comments')
+            .send(postInfo)
+            .expect(400)
+            .then(({body: {msg}}) => {
+                expect(msg).toBe('400 - Bad Request Invalid Data Type')
```

Comment:
We could add a test to check what happens when an additional property is sent :)